src/test/java/com/pubnub/contract/subscribe/steps/ThenSteps.kt

```diff
@@ -0,0 +1,51 @@
+package com.pubnub.contract.subscribe.steps
+
+import com.pubnub.api.models.consumer.pubsub.PNMessageResult
+import com.pubnub.api.models.consumer.pubsub.PNSignalResult
+import com.pubnub.contract.subscribe.state.SubscribeState
+import io.cucumber.java.en.Then
+import org.awaitility.Awaitility.await
+import org.hamcrest.MatcherAssert.assertThat
+import org.hamcrest.Matchers.*
+import org.junit.jupiter.api.Assertions.assertTrue
+import java.util.*
+import java.util.concurrent.TimeUnit
+
+class ThenSteps(
+    private val subscribeState: SubscribeState
+) {
+
+    @Then("I receive 2 messages in my subscribe response")
+    fun I_receive_2_messages_in_my_subscribe_response() {
+        await()
+            .atMost(500, TimeUnit.MILLISECONDS)
+            .until {
+                subscribeState.messages.size == 2
+            }
+    }
+
+    @Then("response contains messages with {string} and {string} message types")
+    fun response_contains_messages_with_message_types(
+        messageTypeOfFirstMessage: String,
+        messageTypeOfSecondMessage: String
+    ) {
+        assertThat(subscribeState.messages.size, equalTo(2))
+        val listOfReceivedMessageTypes = subscribeState.messages.map { it.messageType.value }
+        assertThat(
+            listOfReceivedMessageTypes,
+            containsInAnyOrder(messageTypeOfFirstMessage, messageTypeOfSecondMessage)
+        )
+    }
+
+    @Then("response contains messages without space ids")
+    fun response_contains_messages_without_space_ids() {
+        assertTrue(
+            subscribeState.messages.stream()
+                .allMatch { (it is PNMessageResult || it is PNSignalResult) && it.spaceId == null })
+    }
+
+    @Then("response contains messages with space ids")
+    fun response_contains_messages_with_space_ids() {
+        subscribeState.messages.stream().allMatch { it is PNMessageResult && it.spaceId != null }
```

Comment:
Not an assertion, therefore it will do nothing. Also you're missing signals here

Reply:
Yes, I am missing assertion. But why should I include signals?

Reply:
You include signals in previous step

Reply:
Yes, but step description gives freedom how I implement this, right?

Reply:
Ok, as discussed I removed:
"it is PNMessageResult &&"